src/unistd.rs

```diff
@@ -240,20 +240,23 @@ pub fn getppid() -> Pid {
 /// Set a process group ID (see
 /// [setpgid(2)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/setpgid.html)).
 ///
-/// Set the process group id (PGID) of a particular process.  If a pid of zero
-/// is specified, then the pid of the calling process is used.  Process groups
-/// may be used to group together a set of processes in order for the OS to
-/// apply some operations across the group.
+/// Set the process group id (PGID) of a particular process.  If None
+/// or a pid of zero is specified, then the pid of the calling process
```

Comment:
Do we really want to still support a Pid of 0 here? I'd argue that the semantics would be improved by forcing people o use `None` or a valid `Pid` rather than having two possible values for `None`.

Reply:
I have spent a while thinking about this now, and I don't see a better option than the one here:

* To disallow `0`, we'd need to be able to have an `.into` that can fail. The only way I know to do that would be to use `TryInto`, which isn't stable yet.
* If we disallowed `0` pid arguments, we'd become incompatible with other code that treats `0` as the "self" pid argument (e.g. C code).
* It's a super incompatible change with rust client code, as stuff that passed in `0` pids before no longer can (the program for which I am making this change would be one of them).

So I'm not sure those tradeoffs are worth introducing the additional complexity (and ergonomic setbacks) of a pid-but-not-zero type. Does that make sense?